### PR TITLE
Fix google analytics

### LINKED ISSUE #576: Fix Google Analytics
[`fix-google-analytics`](https://github.com/iFixit/react-commerce/compare/fix-google-analytics)

https://ifixit.slack.com/archives/C01K32B4FPX/p1659483805779779

Google Tag Assistant says it isn't set up right.

Implemented in #485 . QA verified that the js was in the page, but didn't verify that data was sent properly.

### PR DESCRIPTION
**Pull request by masonmcelvain**
- Fixes a bug where page views weren't being tracked properly.
- Follows the Next.js [reference implementation](https://nextjs.org/docs/messages/next-script-for-ga#using-analyticsjs) of Google Analytics. Note that this puts the script in the `<body>`, and it was in the `<head>` before. This doesn't seem to make a difference functionally. I couldn't find a way to use the Next's `<Script>` component in the `<head>`.
- Sends a GA pageview when the Next router indicates that the route has changed.

## QA
- Install the Legacy Google Tag Manager extension on Chrome: https://chrome.google.com/webstore/detail/tag-assistant-legacy-by-g/kejbdjndbnbjgmefkgdddjlbokphdefk?hl=en
- Run react-commerce locally, and set `NEXT_PUBLIC_GA_URL=https://www.google-analytics.com/analytics.js` in `.env.development`.
- Click "Record" in the extension, and load any page of the app.
- When the page is finished loading, click "Stop Recording" in the extension. It should tell you the page was tracked properly. It will warn you that the script is in the `<body>` instead of the `<head>`, but as long as tracking works properly, I don't think this matters too much.

This is what it should look like:
image

CC @sterlinghirsh @davidrans 

Closes #576

### CODE REVIEW

**Review by davidrans — APPROVED** [review]
Nice work!
CR :coffee:

File: frontend/components/analytics/GoogleAnalytics.tsx [hunk 1]
```diff
@@ -1,35 +1,54 @@
...
+import * as React from 'react';
+import Script from 'next/script';
+import { useRouter } from 'next/router';
+
+declare const ga: (command: string, hitType: string, url?: string) => void;
 
 export function GoogleAnalytics() {
+   const router = useRouter();
+   React.useEffect(() => {
+      const handleRouteChange = (url: string) => {
```
Comment (davidrans) [inline frontend/components/analytics/GoogleAnalytics.tsx:11]: Do we need to do this same thing for Matomo?

File: frontend/components/analytics/GoogleAnalytics.tsx [hunk 2]
```diff
@@ -1,35 +1,54 @@
...
+               ga('ifixit.require', 'ec');
+
+               // Enable Remarketing and Advertising Reporting Features in GA
+               ga('ifixit.require', 'displayfeatures');
+
+               ga('ifixit.set', 'page', window.location.pathname);
+               ga('ifixit.send', 'pageview');
+            `}
+         </Script>
+         <Script src={`${GA_URL}`} strategy="afterInteractive" />
```
Comment (masonmcelvain) [inline frontend/components/analytics/GoogleAnalytics.tsx:51]:
```suggestion
         <Script src={GA_URL} strategy="afterInteractive" />
```

File: frontend/components/analytics/GoogleAnalytics.tsx (hunk 1 above)
Comment (masonmcelvain) [reply frontend/components/analytics/GoogleAnalytics.tsx:11]: I'll check if navigation is working, didn't think of that.

**Review by davidrans — APPROVED** [review]
CR :coffee:

File: frontend/components/analytics/GoogleAnalytics.tsx (hunk 1 above)
Comment (masonmcelvain) [reply frontend/components/analytics/GoogleAnalytics.tsx:11]: It doesn't seem like Matomo knows about route changes. I made an issue: https://github.com/iFixit/react-commerce/issues/595

**Comment by deltuh-vee** [comment]
QA 🎬 
Page views are now tracked with Google Analytics.

**Pull request merged**